### PR TITLE
Add infrastructure lambda support

### PR DESCRIPTION
**Pull request by stijn-janssens**
Allow triggering of custom lambda (defined elsewhere) on certain events that could impact your infrastructure
Fix some missing settings values

### CODE REVIEW

File: backend/app/core/aws/refresh_infrastructure_lambda.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+from app.settings import settings
+
+
+class RefreshInfrastructureLambda:
+    def __init__(self):
+        self.enabled = bool(settings.INFRASTRUCTURE_LAMBDA_ARN)
+        self.lambda_arn = settings.INFRASTRUCTURE_LAMBDA_ARN
+        self.logger = getLogger()
+        if self.enabled:
+            self.client = get_client("lambda")
```
Comment (andrii-ivaniuk) [inline backend/app/core/aws/refresh_infrastructure_lambda.py:16]:
```suggestion
    def __init__(self):
        self.lambda_arn = settings.INFRASTRUCTURE_LAMBDA_ARN
        self.logger = getLogger()
        if self.lambda_arn:
            self.client = get_client("lambda")
```
No need to have extra attribute

File: backend/app/core/aws/refresh_infrastructure_lambda.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    def trigger(self) -> dict[str, Any]:
+        if self.enabled:
```
Comment (andrii-ivaniuk) [inline backend/app/core/aws/refresh_infrastructure_lambda.py:19]:
```suggestion
        if self.lambda_arn:
```

File: backend/app/core/aws/refresh_infrastructure_lambda.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            try:
+                response = self.client.invoke(
+                    FunctionName=self.lambda_arn, InvocationType="Event"
+                )
+                self.logger.info(response)
+                return response
+            except ClientError:
+                self.logger.warning("Triggering failed", exc_info=1)
+                return {"status": "triggering failed"}
```
Comment (andrii-ivaniuk) [inline backend/app/core/aws/refresh_infrastructure_lambda.py:28]: Do we use return value somewhere?

Comment (stijn-janssens) [reply backend/app/core/aws/refresh_infrastructure_lambda.py:28]: currently not

**Pull request merged**